Sources/AsyncAlgorithms/Merge/MergeStateMachine.swift

```diff
@@ -41,7 +41,8 @@ struct MergeStateMachine<
             buffer: Deque<Element>,
             upstreamContinuations: [UnsafeContinuation<Void, Error>],
             upstreamsFinished: Int,
-            downstreamContinuation: UnsafeContinuation<Element?, Error>?
+            downstreamContinuation: UnsafeContinuation<Element?, Error>?,
+            cancelled: Bool
```

Comment:
should this just be another state case?

Reply:
Agreed. I think we should just leave it with the current case